agent/canvas.py

```
@@ -660,22 +660,49 @@ def _node_finished(cpn_obj):
+                        tts_sequence = 0
+                        next_audio_sequence = 0
+                        completed_tts = {}
                         sentence_end = re.compile(r"(?:[。！？!?；;\n](?:[”’\"』」】》）)]*)|(?<=[.!?])(?=\s|$))")
                         stream = cpn_obj.output("content")()
 
-                        def _schedule_tts(text):
+                        async def _tts_worker():
+                            while True:
+                                sequence, text = await tts_queue.get()
+                                try:
+                                    audio_binary = await asyncio.to_thread(self.tts, tts_mdl, text)
```

Ruff (0.16.1)

[warning] Function definition does not bind loop variable `tts_mdl`

(B023)